api/types/image_summary.go

```diff
@@ -1,7 +1,74 @@
 package types
 
-// This file was generated by the swagger tool.
-// Editing this file might prove futile when you re-run the swagger generate command
+import specs "github.com/opencontainers/image-spec/specs-go/v1"
+
+// This file was originally generated by the swagger tool.
+//
+// Turned off auto-generation for this file due to limitation in go-swagger to be
+// able to generate an entirely compatible OCI image-spec platform structure.
+//
+// Issues were
+//
+//  a) $ref was auto-generating the structure being referenced
+//  b) When inlining the definition such as follows:
+//
+//      Platforms:
+//        type: "object"
+//        description: "An array containing all platforms supported by the image"
+//        items:
+//          type: "object"
+//          properties:
+//            Architecture:
+//              type: "string"
+//            OS:
+//              description: "operating system"
+//              type: "string"
+//            OSVersion:
+//              description: "operating system version"
+//              type: "string"
+//            OSFeatures:
+//              description: "operating system features"
+//              type: "array"
+//              items:
+//                type: "string"
+//            Variant:
+//               type: "string"
+//            Features:
+//              type: "array"
+//              items:
+//                type: "string"
+//
+// What gets generated is:
+//
+//
+// ImageSummaryPlatformsItems0 image summary platforms items0
+// swagger:model ImageSummaryPlatformsItems0
+//type ImageSummaryPlatformsItems0 struct {
+//
+//	// architecture
+//	Architecture string `json:"Architecture,omitempty"`
+//
+//	// features
+//	Features []string `json:"Features"`
+//
+//	// operating system
+//	OS string `json:"OS,omitempty"`
+//
+//	// operating system features
+//	OSFeatures []string `json:"OSFeatures"`
+//
+//	// operating system version
+//	OSVersion string `json:"OSVersion,omitempty"`
+//
+//	// variant
+//	Variant string `json:"Variant,omitempty"`
+//}
+//
+// Note that there's no means to alias a json name (eg in the OCI spec, it's
+// os.version and os.features, not OSVersion and OSFeatures). Also the name
+// `ImageSummaryPlatformsItems0` is fixed and no way to give it a 'meaninful'
+// name. Ideally, there's no need for this structure, as it should just import
```

Comment:
Looks like there's `x-go-name` at least for the model; https://github.com/go-swagger/go-swagger/pull/713 but not sure if a similar thing exist for individual properties; @dnephin may know that

Reply:
Even if that works, it doesn't solve the `OSFeatures` to `os.features` problem. In reality, unless the version of go-swagger in use has a way to directly pull in an already defined struct (ie the Platform struct in the image-spec), we're kind of re-inventing the wheel...

Reply:
Yup, understood that (was just curious if it was possible in general for other cases)

I actually was looking a while back at the option to reverse the direction, and generate the swagger from the code, but that would also mean that updating docs requires code-changes (e.g. if we want to have a nicer description for the docs)

Reply:
`x-go-name` is supposed to work.

The source of truth should really be `api/types`. It's good that we mirror the `specs.Platform`, but if we use the imported struct the type could change without us really knowing about the change, and we could break API compatibility. 

It's also important to be consistent with other fields. The swagger generated type is a lot more consistent, so I would say don't use `x-go-name` at all. `os.features` is not like anything else we have in the API.

It's going we match the same fields, but I don't think we should use the imported struct.

Also why an array of pointers?

Reply:
> Also why an array of pointers?

See comment https://github.com/moby/moby/pull/34924#issuecomment-331308527 from @stevvooe 

@dnephin - I'm not quite sure what you're saying in the rest of it. Are you saying hard-code something that's identical to the OCI image spec `Platform`, but just not use the import?

Reply:
I don't see anything in that comment about pointers, could you elaborate?

I think it should have the same "form" as the OCI  Platform spec, but the field names should be consistent with the other fields in the API. So `OSFeatures` seems more appropriate than `os.features`.

Reply:
@dnephin We are following the OCI platform structure from image spec.

`os.features` was deprecated, so it actually shouldn't be defined here.